### ISSUE TITLE
Difference in embedding when comparing SentenceTransformer vs TEI 0.4.0 and up for multilingual-e5-large

### ISSUE THREAD

**Issue by gerritvd** [huggingface/text-embeddings-inference#94]
### System Info

Comparing SentenceTransformer output with TEI 0.2.2, 0.3.0, 0.4.0, and 0.5.0

### Information

- [x] Docker
- [ ] The CLI directly

### Tasks

- [x] An officially supported command
- [ ] My own modifications

### Reproduction

I deployed the latest version of **multilingual-e5-large** on the following versions of TEI

1. v0.2.2
2. v0.3.0
3. v0.4.0
4. v0.5.0

They are ran using:
```
        # Replace with different versions
        image: ghcr.io/huggingface/text-embeddings-inference:89-0.3.0
        command: [ "text-embeddings-router" ]
        args:
          - "--port"
          - "7860"
          - "--dtype"
          - "float16"
          - "--model-id"
          - "/mnt/models/hub/models--intfloat--multilingual-e5-large/snapshots/9f78368af0062735ba99812349c562316e29f719"
```

I then generate an embedding using SentenceTransformer:
```
from sentence_transformers import SentenceTransformer
model = SentenceTransformer("intfloat/multilingual-e5-large", device='cpu')
payload = [' '.join(["Car"]*510)]
st_result = model.encode(payload)
```
and I send that same payload to the other 4 backends, enabling normalize and truncate where supported (although truncation is not happening).
I then calculate the cosine similarity between all these vectors:
```
from sklearn.metrics.pairwise import cosine_similarity
cos_sim = cosine_similarity(st_result, tei_result)
```

### Expected behavior

I expected the cosine similarities to be very close to 1 for all these values. However, I'm recording the following values:

st_result vs:
1. v0.2.2 = 0.9999881 (close enough)
2. v0.3.0 = 0.9999881 (close enough)
3. v0.4.0 = 0.95674335 (looks too far off)
4. v0.5.0 = 0.95674335 (looks too far off)

Is there a clear explanation of why this is happening? I did not a similar difference when using gte-large (ST vs TEI)

**Comment by gerritvd** [comment]
Thanks for the quick turn around!

**Comment by scriptator** [comment]
Sadly, I have to report that the problem is still not fully fixed. You only have to add a trailing space and it breaks again. My test script

```python
e5 = SentenceTransformer("intfloat/multilingual-e5-large")
headers = {'Content-Type': 'application/json'}

url_e5 = 'http://localhost:33200/embed'

texts = [
    ' '.join(["Car"]*510),
    ' '.join(["Car"]*507),
    ' '.join(["Car"]*507) + ' ',
    ' '.join(["Car"]*507) + ' \n',
    ' '.join(["Car"]*507) + ' \n ',
]

data = {
  "inputs": texts,
  "normalize": True,
  "truncate": False,
}

response_e5 = httpx.post(url_e5, headers=headers, json=data, timeout=60)
tei_vectors_e5 = response_e5.json()

native_vectors_e5 = e5.encode(texts)

print(cos_sim(tei_vectors_e5[0], native_vectors_e5[0]))
print(cos_sim(tei_vectors_e5[1], native_vectors_e5[1]))
print(cos_sim(tei_vectors_e5[2], native_vectors_e5[2]))
print(cos_sim(tei_vectors_e5[3], native_vectors_e5[3]))
print(cos_sim(tei_vectors_e5[4], native_vectors_e5[4]))
```

prints the following response

```
0.9999883188452392    // original example from this issue --> ok
0.9999884754659525    // original example with just 507 cars --> ok
0.9853260777816594    // add a trailing whitespace --> bad
0.9792360657039959    // add a trailing whitespace and a newline --> even worse
0.9760671020914671    // add a trailing whitespace and a newline and a whitespace --> even worse
```

Environment:
- TEI: Docker CPU build from commit 6395a7a29624bb7199acf58df086005bb250f35e
- Local:
  - Python 3.11.7
  - sentence-transformers==2.2.2
  - tokenizers==0.15.0

**Comment by scriptator** [comment]
By the way, removing the whole special handling that was introduced due to https://github.com/huggingface/tokenizers/pull/1357 I get all similarities to 0.99998.

![image](https://github.com/huggingface/text-embeddings-inference/assets/6749507/be42a12b-6522-4894-b1fa-07871ee27fd7)

Could it be that this whole block is not necessary any more because the upstream issue was fixed? I don't understand the subject well enough to assess this on my own. I also don't want to introduce further regressions with other models.

**Comment by scriptator** [comment]
@OlivierDehaene did you already have the chance to think about my suggestion?

**Comment by OlivierDehaene** [comment]
This code was removed on main.

**Comment by scriptator** [comment]
If anybody else should be searching the commit: https://github.com/huggingface/text-embeddings-inference/commit/9d35f82cf3c29afc981fcdf6e04557f43a1aebae#diff-c3259a23eb2e9e4e39f9a5226354b99508e42b05a158036dc482397410becd52
